### PR TITLE
Test: Filter multi format impressions

### PR DESCRIPTION
**Pull request by marki1an**
Tests for https://github.com/prebid/prebid-server-java/pull/2780

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/model/config/PreferredMediaType.groovy [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import com.fasterxml.jackson.databind.PropertyNamingStrategies
+import com.fasterxml.jackson.databind.annotation.JsonNaming
+import groovy.transform.ToString
+import org.prebid.server.functional.model.response.auction.MediaType
+
+@ToString(includeNames = true, ignoreNulls = true)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy)
+class PreferredMediaType {
+
+    MediaType generic
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/model/config/PreferredMediaType.groovy:12]: It's per-bidder? Can we express it in a somewhat different way? Or at least make it obvious that it's a bidder. Dumb option would be to add some other bidders here. Other option is to make `AccountAuctionConfig` field for this a map of bidder to preferred media type.

File: src/test/groovy/org/prebid/server/functional/model/request/auction/Bidder.groovy [hunk 2]
```diff
@@ -19,6 +19,8 @@ class Bidder {
     @JsonProperty("appnexus")
     AppNexus appNexus
     Openx openx
+    Bidder bidder
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/model/request/auction/Bidder.groovy:22]: You just lost me here. What is the point of recursion here?

File: src/test/groovy/org/prebid/server/functional/model/request/auction/Prebid.groovy [hunk 3]
```diff
@@ -36,6 +35,7 @@ class Prebid {
     Boolean createTids
     Sdk sdk
     List<AdServerTargeting> adServerTargeting
+    Bidder bidders
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/model/request/auction/Prebid.groovy:38]: We have the same object in several places?

**Review by Net-burst — CHANGES_REQUESTED** [review]

File: src/test/groovy/org/prebid/server/functional/tests/FilterMultiFormatSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+import org.prebid.server.functional.model.request.auction.Bidder
+import org.prebid.server.functional.model.request.auction.Native
+
+import static org.prebid.server.functional.model.response.auction.ErrorType.GENERIC
+import static org.prebid.server.functional.model.response.auction.MediaType.AUDIO
+import static org.prebid.server.functional.model.response.auction.MediaType.BANNER
+
+class FilterMultiFormatSpec extends BaseSpec {
+
+    def "PBS should response with all requested media type when default adapters multi format true in config and at account level specified preferred media type"() {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/FilterMultiFormatSpec.groovy:20]: "PBS should response with all requested media type when default adapters multi format true in config and at account level specified preferred media type" -> "PBS should respond with all requested media types when default adapters multi format is true in config and preferred media type specified at account level"

There is a very similar grammar errors in all test names. Please proofread and correct them.

File: src/test/groovy/org/prebid/server/functional/model/bidder/Generic.groovy [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package org.prebid.server.functional.model.bidder
 
 import com.fasterxml.jackson.annotation.JsonProperty
+import org.prebid.server.functional.model.response.auction.MediaType
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/model/bidder/Generic.groovy:4]: unused import

File: src/test/groovy/org/prebid/server/functional/tests/FilterMultiFormatSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+        }
+
+        when: "PBS processes auction request"
+        def bidResponse = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS should make bidder request"
+        assert bidder.getBidderRequests(bidRequest.id)
+
+        and: "Bid response shouldn't contain warning"
+        assert !bidResponse.ext?.warnings
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/FilterMultiFormatSpec.groovy:328]:
```
If the request does not contain a preferred type:
The bidder will not be called and a warning will be returned.
```
https://github.com/prebid/prebid-server/issues/2711#issue-1677787288

`if a preferred mediatype is not specified, there's no default. The bid adapter may choke.`

https://github.com/prebid/prebid-server/issues/2711#issuecomment-1518069774

**Review by osulzhenko — CHANGES_REQUESTED** [review]

**Pull request merged**